Editor/UI/ProjectAuditorWindow.cs

```diff
@@ -150,6 +152,9 @@ void OnEnable()
             UpdateAssemblySelection();
             Profiler.EndSample();
 
+            m_SettingsProvider = new ProjectAuditorSettingsProvider();
```

Comment:
Oh, right, it is both a minor optimization since I need that instance (for the UI logic) and mainly an idea to use "injection":

The window logic deals with the same SettingsProvider anyway (caches the settings assets; may change the settings and possibly EditorPrefs, if we don't use the default).

If we further provide it to the later instantiated ProjectAuditor, that one doesn't need a new 2nd instance and run similar logic (caches the settings assets, choses the last one from EditorPrefs or default).

That way we also follow an injection pattern, where from the calling code we know what we provide (inject) what is to be used for analysis, similar to the unit tests where I do the same.